src/backend/cpu/queue.hpp

```diff
@@ -48,32 +49,45 @@ typedef async_queue queue_impl;
 namespace cpu {
 
 /// Wraps the async_queue class
-class queue {
+class queue
+{
 public:
-  queue()
-    : sync_calls( __SYNCHRONOUS_ARCH == 1 || getEnvVar("AF_SYNCHRONOUS_CALLS") == "1") {}
-
-  template <typename F, typename... Args>
-  void enqueue(const F func, Args... args) {
+    queue()
+        :
+        count(0),
+        sync_calls( __SYNCHRONOUS_ARCH == 1 || getEnvVar("AF_SYNCHRONOUS_CALLS") == "1")
+    {}
 
-    if(sync_calls) { func( args... ); }
-    else           { aQueue.enqueue( func, args... ); }
+    template <typename F, typename... Args>
+    void enqueue(const F func, Args... args)
+    {
+        count++;
+        if(sync_calls) { func( args... ); }
+        else           { aQueue.enqueue( func, args... ); }
 #ifndef NDEBUG
-    sync();
+        sync();
+#else
+        if (checkMemoryLimit() || count >= 25) {
```

Comment:
This is a work around for now. We should probably remove it entirely by having a proper solution in the future.